daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/Parser.scala

```diff
@@ -241,8 +244,14 @@ final class SeqCompParser(context: RuntimeData, val childParsers: Vector[Parser]
     while (i < numChildParsers) {
       val parser = childParsers(i)
       parser.parse1(pstate)
-      if (pstate.processorStatus ne Success)
+      if (pstate.processorStatus ne Success) {
+        if (testAssert.nonEmpty) {
+          testAssert.foreach { ap =>
+            pstate.withTempSuccess(ap.parse1)
+          }
+        }
```

Comment:
Ah, I guess I now see why it's okay for `childrenParsers` to also include the assertion parsers. If all the non-assert parsers succeed then we'll don't evaluate the assertion parsers.

However, I think there is still an issue. Say all the non-assertion parsers succeed, then we start evaluating the assertion parsers, and assume one of them fails. The `pstate.processorStatus ne Success` will be true, and then we'll evaluate all the assertion parsers, including ones we've already evaluated.

So I think this approach still feels like it needs a tweak. Almost feels like the assert parsers need to be completely separate from the other parsers or something.

Reply:
I think I Have addressed this with the changes I mentioned above by separating out the discriminator statements and running them after the rest of the sequence body.  Nothing should be getting run twice anymore.